feature/autodiscovery/demo/src/main/kotlin/app/k9mail/autodiscovery/demo/DemoAutoDiscovery.kt

```diff
@@ -0,0 +1,50 @@
+package app.k9mail.autodiscovery.demo
+
+import app.k9mail.autodiscovery.api.AutoDiscovery
+import app.k9mail.autodiscovery.api.AutoDiscoveryResult
+import app.k9mail.autodiscovery.api.AutoDiscoveryRunnable
+import app.k9mail.autodiscovery.api.IncomingServerSettings
+import app.k9mail.autodiscovery.api.OutgoingServerSettings
+import app.k9mail.core.common.mail.EmailAddress
+import app.k9mail.core.common.mail.toDomain
+import com.fsck.k9.mail.AuthType
+import com.fsck.k9.mail.ConnectionSecurity
+import com.fsck.k9.mail.ProtocolType
+import com.fsck.k9.mail.ServerSettings
+import com.fsck.k9.mail.protocolType
+
+class DemoAutoDiscovery : AutoDiscovery {
+    override fun initDiscovery(email: EmailAddress): List<AutoDiscoveryRunnable> {
+        val domain = email.domain.toDomain()
+
+        return listOf(
+            AutoDiscoveryRunnable {
+                if (domain.value == "example.com") {
```

Comment:
If we use the TLDs `example` and/or `test`, we make sure the app never accidentally connects to a remote server because these domains are guaranteed not to resolve on the public internet (unlike `example.com`).

Reply:
Where does it connect to the internet?

Reply:
Aside from Autoconfig (that isn't used for demo accounts), I don't think the app currently uses the domain part of the email address to connect to a remote server. But that might change at some point in the future.

I think of `example.com` as the old way of having an example domain. Using `${something}.example` as the new way. Both work fine. Feel free to ignore this suggestion.